agent/src/agentclient.rs

```diff
@@ -312,119 +400,48 @@ impl<T: APIServerClient> BrupopAgent<T> {
-                        event!(Level::INFO, "Preparing update");
-                        prepare().await.context(agentclient_error::UpdateActions {
-                            action: "Prepare".to_string(),
-                        })?;
-
-                        self.cordon_and_drain().await?;
-                    }
-                    BottlerocketShadowState::PerformedUpdate => {
-                        event!(Level::INFO, "Performing update");
-                        update().await.context(agentclient_error::UpdateActions {
-                            action: "Perform".to_string(),
-                        })?;
-                    }
-                    BottlerocketShadowState::RebootedIntoUpdate => {
-                        event!(Level::INFO, "Rebooting node to complete update");
-
-                        if running_desired_version(&bottlerocket_shadow_spec).await? {
-                            // previous execution `reboot` exited loop and did not update the custom resource
-                            // associated with this node. When re-enter loop and finished reboot,
-                            // try to update the custom resource.
-                            let update_node_status = match self
-                                .gather_system_metadata(bottlerocket_shadow_spec.state.clone())
-                                .await
-                            {
-                                Ok(update_node_status) => update_node_status,
-                                Err(_) => {
-                                    // Errors gathering system metadata are ignored (and also logged by `gather_system_metadata()`).
-                                    event!(Level::WARN, "An error occurred when gathering system metadata. Restarting event loop");
-                                    sleep(AGENT_SLEEP_DURATION).await;
-                                    continue;
-                                }
-                            };
-                            if self
-                                .update_metadata_custom_resource(update_node_status)
-                                .await
-                                .is_err()
-                            {
-                                // Errors updating BottlerocketShadow are ignored (and also logged by `update_metadata_custom_resource()`).
-                                event!(Level::WARN, "An error occurred when updating BottlerocketShadow. Restarting event loop");
-                                sleep(AGENT_SLEEP_DURATION).await;
-                                continue;
-                            };
-                        } else {
-                            boot_update()
-                                .await
-                                .context(agentclient_error::UpdateActions {
-                                    action: "Reboot".to_string(),
-                                })?;
+            match self.handle_update(&bottlerocket_shadow).await {
```

Comment:
nit: We should pick a more particular name here, maybe `handle_state_transition`?